content/courses/Ada_For_The_C_Embedded_Developer/chapters/04_Embedded.rst

```diff
@@ -231,9 +231,209 @@ Understanding Bare-Metal Environment
 Dealing with Absence of FPU with Fixed Point
 --------------------------------------------
 
-.. todo::
+Many numerical applications typically use floating-point types to compute
+values. However, in some platforms, a floating-point unit may not be available.
+Other platforms may have a floating-point unit, but using it in certain
+numerical algorithms can be prohibitive in terms of performance. For those
+cases, fixed-point arithmetic can be a good alternative.
 
-    Complete section!
+The difference between fixed-point and floating-point types might not be so
+obvious when looking at this code snippet:
+
+[Ada]
+
+.. code-block:: ada
+
+    --  [...]
+
+       Float_Value : Float := 0.25;
+       Fixed_Value : Fixed := 0.25;
+
+    begin
+
+       Float_Value := Float_Value + 0.25;
+       Fixed_Value := Fixed_Value + 0.25;
+
+       Put_Line ("Float_Value = " & Float'Image (Float_Value));
+       Put_Line ("Fixed_Value = " & Fixed'Image (Fixed_Value));
+
+    --  [...]
+
+In this example, the application will show the value 0.5 for both
+:ada:`Float_Value` and :ada:`Fixed_Value`.
+
+The major difference between floating-point and fixed-point types is in the way
+the values are stored. Values of ordinary fixed-point types are, in effect,
+scaled integers. The scaling used for ordinary fixed-point types is defined by
+the type's :ada:`small`, which is derived from the specified :ada:`delta` and,
+by default, is a power of two. Therefore, ordinary fixed-point types are
+sometimes called binary fixed-point types. In that sense, ordinary fixed-point
+types can be thought of being close to the actual representation on the
+machine. In fact, ordinary fixed-point types make use of the available integer
+shift instructions, for example.
+
+Another difference between floating-point and fixed-point types is that Ada
+doesn't provide standard fixed-point types. While the Ada standard specifies
```

Comment:
(other than type Duration)

Reply:
Thanks for the hint. I've just added it as an exception to the rule. :-)